pkg/spec/spec.go

```diff
@@ -31,14 +31,17 @@ type Function struct {
 
 // FunctionSpec contains func specification
 type FunctionSpec struct {
```

Comment:
Of everything in this PR, changes to this struct probably deserve the most thought/discussion, since this is user-facing and changes here directly affect compatibility.

So: In a change that strives to add multi-file support - I'm surprised we're only adding the ability to describe a single file.  More generally, since we already have two potential sources (inline and from minio) and are discussing others, I think we should use a discriminated enum, so that we can add additional methods to fetch content in the future (eg: from other object stores, arbitrary URLs, git, arbitrary volume Mounts, etc, etc) with minimal drama.

Reply:
> I'm surprised we're only adding the ability to describe a single file

The ability to use more than one file is given supporting a single Zip file. That's why this description only describes a single file.

> More generally, since we already have two potential sources (inline and from minio) and are discussing others, I think we should use a discriminated enum, so that we can add additional methods to fetch content in the future\

That's was my intention with the "ContentType" field. That plus the `function` field allow us to specify a git URL, a  S3 path, a Volume identifier...

Reply:
[Out of PR's scope] So is it time for us to start thinking of creating interface for kubeless ? Like we want to plug data sources that contain function's content (s3, minio, local volume, etc). 

Btw, the name of field - `function` - is not a good name as it confuses what exactly it is used for.

Reply:
> [Out of PR's scope] So is it time for us to start thinking of creating interface for kubeless ? Like we want to plug data sources that contain function's content (s3, minio, local volume, etc).

Yes, the `Function` it is an interface so we need to treat like that

> Btw, the name of field - function - is not a good name as it confuses what exactly it is used for.

I agree that it may be not the best name but we need to leave it like that for the moment to avoid breaking changes. Once we increase the major version we can rethink about the best name.